### PR TITLE
Adds support for "realistic" public elevators (elevator doors, elevator panel, etc)

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

<details> 

<summary> UI! </summary>

image

</details>
<details> 

<summary> Dimmers! </summary>

image

</details> 
<details> 

<summary> Warnings! </summary>

image

</details> 

This PR adds support for mappers to add more "public facing" elevators instead of the industrial lifts. 

- Elevators can now connect to doors and shutters, allowing elevator doors that open and shut where needed. 
- Elevators can now have elevator panels that show all destinations 
- Elevators can now optionally be a bit "nicer", showing a warning to the level below when it's about to land, as well as severely crippling instead of gibbing. 
   - Elevators which opt to use this can be emagged to return the original gibbing and no warning behavior. 
- Elevators now have sounds when travelling, and have a travel time. 

I put one of these elevators on multizdebug for testing funsies. 

## Why It's Good For The Game

Mappers have long sense been struggling to have a more "realistic" / "authentic" elevator experience for their maps
As it stands, elevators are extremely barebones and drives people away from wanting to use it
This gives them a lot more support and makes them more versatile, which is neat. 

## Changelog

:cl: Melbert
add: Added support for elevator doors. 
add: Elevators now have travel time, and make sound when moving. Emagging the elevator button will speed them up.
add: Added support for less-than-lethal elevators, which crush your bones instead of gibbing you. Emagging the elevator button will revert them to gibbing. 
add: Added support for elevators to warn people below that they're moving down. Emagging the elevator button will disable this. 
add: Added elevator panels, and support for using elevators without a radial. 
qol: Elevator call buttons and the trapdoor now use balloon alerts.
/:cl:

### CODE REVIEW

**Comment by Stalkeros2** [comment]
Making people hear the generic elevator music should be a must have.

**Comment by Farquaar** [comment]
Surprised we don't have this already, considering how many other servers have something similar. Great addition

**Comment by DAKKA-WAAAGH** [comment]
NOW! MERGE THIS NOW!

**Comment by stylemistake** [comment]
@MrMelbert you should improve the theme so that disabled buttons appear legible.

**Review by stylemistake — APPROVED** [review]
Nice

**Comment by AnturK** [comment]
But does it play elevator music ?

**Comment by DrDiasyl** [comment]
Finally! Will be great to see this getting implemented in the most if not all maps with multiple Z levels! Also probably you should make some unique shutters, which would fit the elevator style.

File: code/modules/industrial_lift/elevator_button.dm [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+		return FALSE
+
+	// We can't call an elevator that's moving. You may say "you totally can do that", but that's not modelled
+	if(lift.controls_locked == LIFT_PLATFORM_LOCKED)
+		loc.balloon_alert(activator, "elevator is moving!")
+		return FALSE
+
+	// We can't call an elevator if it's already at this destination
+	var/obj/structure/industrial_lift/prime_lift = lift.lift_platforms[1]
+	if(prime_lift.z == loc.z)
```
Comment (Kylerace) [inline code/modules/industrial_lift/elevator_button.dm:97]: this is also potentially ambiguous for multiz lifts but there doesnt exist a solution for this presently

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/modules/industrial_lift/lift_master.dm [hunk 2]
```diff
@@ -280,14 +277,195 @@ GLOBAL_LIST_EMPTY(active_lifts_by_type)
...
+		addtimer(CALLBACK(src, .proc/open_lift_doors, destination.z), door_duration, TIMER_UNIQUE)
+
+	if(direction != DOWN)
+		return
+
+	// Show warning signs if we're going downwards, to avoid crushing and peril
+	for(var/obj/structure/industrial_lift/going_to_move as anything in lift_platforms)
+		if(!going_to_move.warns_on_down_movement)
+			continue
+		var/turf/below_us = get_step_multiz(going_to_move, DOWN)
```
Comment (Kylerace) [inline code/modules/industrial_lift/lift_master.dm:361]: you cant check just 1 turf per lift because lifts are potentially multi tile

**Pull request merged**